src/libs/Violations/ViolationsUtils.ts

```diff
@@ -49,31 +56,55 @@ function getTagViolationsForSingleLevelTags(
 }
 
 /**
- * Calculates some tag levels required and missing tag violations for the given transaction
+ * Calculates missing tag violations for policies with dependent tags,
+ * by returning one per tag with its corresponding tagName in the data
  */
-function getTagViolationsForMultiLevelTags(
-    updatedTransaction: Transaction,
-    transactionViolations: TransactionViolation[],
-    policyRequiresTags: boolean,
-    policyTagList: PolicyTagList,
-): TransactionViolation[] {
+function getTagViolationsForDependentTags(policyTagList: PolicyTagList, transactionViolations: TransactionViolation[], tagName: string) {
+    const tagViolations = [...transactionViolations];
+
+    if (!tagName) {
+        Object.values(policyTagList).forEach((tagList) =>
+            tagViolations.push({
+                name: CONST.VIOLATIONS.MISSING_TAG,
+                type: CONST.VIOLATION_TYPES.VIOLATION,
+                data: {tagName: tagList.name},
+            }),
+        );
+    } else {
+        const tags = TransactionUtils.getTagArrayFromName(tagName);
+        if (Object.keys(policyTagList).length !== tags.length || tags.includes('')) {
+            tagViolations.push({
+                name: CONST.VIOLATIONS.ALL_TAG_LEVELS_REQUIRED,
+                type: CONST.VIOLATION_TYPES.VIOLATION,
+                data: undefined,
+            });
+        }
+    }
+
+    return tagViolations;
+}
+
+/**
+ * Calculates missing tag violations for policies with independent tags,
+ * by returning one per tag with its corresponding tagName in the data
```

Comment:
```suggestion
 * Calculates missing tag violations for policies with independent tags
```

The second part of this comment isn't accurate. For SOME_TAG_LEVELS_REQUIRED we only return 1 violation, not 1 per tag

Reply:
@cead22 Good catch! Thank you 😄